targon/internal/targon/callbacks.go

```diff
@@ -339,6 +339,43 @@ func setWeights(v *boilerplate.BaseChainSubscriber, c *Core, h types.Header) {
 		"scores",
 		fmt.Sprintf("%+v", scores),
 	)
+
+	// Store emissions data to MongoDB
+	if c.Deps.Mongo != nil && c.EmissionPool != nil && c.TaoPrice != nil {
```

Comment:
why is this in getWeights? If I name a function `getWeights` then it should only get the weights, not cause side effects.